### PR TITLE
Avoid NPE when tailer is closed in parallel

### PR DESCRIPTION
**Pull request by alamar**
fixes OpenHFT/Chronicle-Queue#1059

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java [hunk 1]
```diff
@@ -261,6 +261,9 @@ ScanResult moveToIndex(@NotNull final ExcerptContext ec, final long index) {
     private ScanResult moveToIndexFromTheStart(@NotNull ExcerptContext ec, long index) {
         try {
             Wire wire = ec.wire();
+            if (wire == null)
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java:264]: is it possible to add a test that breaks before this change?

Comment (alamar) [reply src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java:264]: It is `TestUninstallableEventLoop` (on repeat, before the ChronicleEnterprise/Chronicle-FIX#1053)

**Comment by alamar** [comment]
@JerryShea Please check, I've added test.

File: src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java [hunk 2]
```diff
@@ -645,11 +646,15 @@ private ScanResult moveToIndexResult0(final long index) {
...
         index(index);
         final ScanResult scanResult = this.store().moveToIndexForRead(this, sequenceNumber);
-        final Bytes<?> bytes = privateWire().bytes();
         switch (scanResult) {
             case FOUND:
-                state = FOUND_IN_CYCLE;
-                moveToState.onSuccessfulLookup(index, direction, bytes.readPosition());
+                Wire privateWire = privateWire();
+                if (privateWire == null)
+                    state = END_OF_CYCLE;
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/queue/impl/single/StoreTailer.java:656]: we should javadoc TailerState!

**Review by JerryShea — APPROVED** [review]
LGTM

**Review by peter-lawrey — APPROVED** [review]

**Pull request merged**